Improve readme

StevenXL opened the pull request:
A few updates on the commit. No radical changes - just added some links, a small "gif" to get folks excited about the product, and moved some sections around. (For example, moved the dependencies section further down the page to de-emphasize them).

No pride of authorship, so please change these as you see fit. Great program - I'm looking forward to using it many times as I learn Python.

sebastinas commented:
Please squash the commits together.

sebastinas commented on README.rst:
```diff
@@ -2,166 +2,159 @@
+
+* `requests[security]`_ 
+
+If you have problems installing cffi which is needed by pyOpenSSL, please take a look at `cffi docs`_.
```
I think we should refer to the homepage here instead. It's easier to document all installation issues there than keep this section up-to-date.

sebastinas commented on README.rst:
```diff
@@ -170,3 +163,17 @@ Documentation
+.. _community: http://docs.bpython-interpreter.org/community.html
+.. _Robert Anthony Farrell: robertanthonyfarrell@gmail.com
+.. _bpython: http://www.bpython-interpreter.org/
+.. _requests[security]: https://pypi.python.org/pypi/requests/
```
Why is this a link and the other dependencies are not?

sebastinas commented:
Please wrap lines at 80 characeters.

StevenXL commented:
I've squashed the commits, wrapped lines at 80 characters, and removed a few links to make the items that have link and don't have link more consistent.

sebastinas commented on README.rst:
```diff
@@ -2,12 +2,87 @@
+
+.. code-block:: bash
+
+    $ sudo pip install bpython
```
Please do not recommend this. Never.

StevenXL replied:
@sebastinas can you be a bit more explicit about what I should not recommend? Is is the fact that we are using the `sudo` command? I have no problem changing this, but I don't understand the issue, so I don't know what to change it to.

sebastinas replied:
`sudo` is problematic here. See http://notes.pault.ag/debian-python/ for some background. But in short: `sudo pip install` will break horribly if you also install Python modules with a package manager.

sebastinas commented on README.rst:
```diff
@@ -33,140 +106,69 @@ bpython-urwid
- * cyan is swapped with yellow
+============
+The curses used has a bug where the colours are displayed incorrectly: * red  is
+swapped with blue * cyan is swapped with yellow
```
Please keep the list here.

StevenXL replied:
Will do - this was a typo from reformatting the doc to wrap at 80 chars. I thought I caught them all. I'll change this tonight (unfortunately, real work beckons).

StevenXL replied:
done

sebastinas commented on README.rst:
```diff
@@ -33,140 +106,69 @@ bpython-urwid
+.. _cffi docs: https://cffi.readthedocs.org/en/release-0.8/#macos-x
+.. _issues tracker: http://github.com/bpython/bpython/issues/
+.. _pip: https://pip.pypa.io/en/latest/index.html
+.. _project homepage: http://bpython-interpreter.org/community
```
Remove the community part.

StevenXL replied:
done

sebastinas commented on README.rst:
```diff
@@ -33,140 +106,69 @@ bpython-urwid
+Contact & Contributing
+======================
+I hope you find it useful and please feel free to submit any bugs/patches
+suggestions to `Robert Athony Farrell`_ or place them on the github
```
@bobf: Do you still want to receive bugs/patches per mail or should we just refer to the bug tracker and the mailing list?

sebastinas commented on README.rst:
```diff
@@ -33,140 +101,69 @@ bpython-urwid
+==========
+Known Bugs
+==========
+For known bugs please see bpython's `issues tracker`_ at github.
```
It's GitHub. Please also refer to http://bpython-interpreter.org/issues.html (should go live later today).

sebastinas replied:
That should be http://bpython-interpreter.org/known-issues-and-faq.html.

sebastinas commented on README.rst:
```diff
@@ -18,14 +93,7 @@ Dependencies
-If you have problems installing cffi which is needed by pyOpenSSL,
-please take a look at https://cffi.readthedocs.org/en/release-0.8/#macos-x.
-
+Please refer to our documentation for further details visite our `homepage`_.
```
There is something missing here.

sebastinas commented:
Once my two remaining comments are fixed, I think this PR is ready to be merged. Thanks for working on this, @StevenXL.

StevenXL commented:
@sebastinas sorry for the amount of time this has taken. I've updated the PR to reflect your suggestions. Let me know if there are other items that need work. I hope I merged the commits correctly.
